substratest/factory.py

```diff
@@ -16,21 +16,32 @@
 
 DEFAULT_SUBSTRATOOLS_VERSION = '0.6.0'
 
-# TODO improve opener get_X/get_y methods
-# TODO improve metrics score method
-
-DEFAULT_OPENER_SCRIPT = """
+DEFAULT_OPENER_SCRIPT = f"""
+import csv
 import json
+import os
 import substratools as tools
 class TestOpener(tools.Opener):
     def get_X(self, folders):
-        return folders
+        res = []
+        for folder in folders:
+            with open(os.path.join(folder, '{DEFAULT_DATA_SAMPLE_FILENAME}'), 'r') as f:
+                reader = csv.reader(f)
+                for row in reader:
+                    res.append(int(row[0]))
+        return res  # returns a list of 1's
     def get_y(self, folders):
-        return folders
-    def fake_X(self, n_samples=None):
-        return 'fakeX'
-    def fake_y(self, n_samples=None):
-        return 'fakey'
+        res = []
+        for folder in folders:
+            with open(os.path.join(folder, '{DEFAULT_DATA_SAMPLE_FILENAME}'), 'r') as f:
+                reader = csv.reader(f)
+                for row in reader:
+                    res.append(int(row[1]))
+        return res  # returns a list of 2's
+    def fake_X(self):
+        return [1]
+    def fake_y(self):
+        return [2]
```

Comment:
There is a new n_samples arg for the fake_X and fake_y method, but this means the code could become 
```python
    def fake_X(self, n_samples=1):
        return [1] * n_samples
    def fake_y(self, n_samples=1):
        return [2] * n_samples
```

Reply:
nice! done